LX/src/main/java/heronarts/lx/LXSnapshot.java

```diff
@@ -0,0 +1,243 @@
+package heronarts.lx;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.modulator.LXModulator;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+
+public class LXSnapshot extends LXComponent {
+
+  public Integer snapshotID;
```

Comment:
private int index;

Shouldn't be public, and no need for heavy Integer type. I'd look at LXModulator.java and follow the naming conventions for setIndex() / getIndex() etc.